### PR TITLE
replaced MSVC variable with WIN32

### PR DESCRIPTION
**Pull request by ericLemanissier**
solves https://github.com/STEllAR-GROUP/hpx/issues/1756

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -52,7 +52,7 @@ set(HPX_LIBRARY_VERSION "${HPX_VERSION}")
 set(HPX_SOVERSION ${HPX_MAJOR_VERSION})
 set(HPX_PACKAGE_NAME HPX)
 
-if(MSVC)
+if(WIN32)
```
Comment (sithhell) [inline CMakeLists.txt]: This is only needed when generating solutions for visual studio. Is this the case?

File: CMakeLists.txt [hunk 2]
```diff
@@ -820,7 +820,8 @@ if(NOT MSVC)
   endif()
 endif()
 
-if(MSVC)
+if(WIN32)
+  if(MSVC)
```
Comment (sithhell) [inline CMakeLists.txt]: Proper indendation would be nice here.

File: CMakeLists.txt [hunk 3]
```diff
@@ -791,7 +791,7 @@ hpx_check_for_cxx14_std_integer_sequence(
 hpx_check_for_cxx14_std_result_of_sfinae(
   DEFINITIONS HPX_HAVE_CXX14_STD_RESULT_OF_SFINAE)
 
-if(NOT MSVC)
+if(NOT WIN32)
```
Comment (sithhell) [inline CMakeLists.txt]: Do I assume correctly that mingw doesn't use the POSIX headers but the windows API exclusively?

Comment (ericLemanissier) [inline CMakeLists.txt]: I'm not a specialist about this, but I would say MinGW has very few POSIX headers, but it has at least POSIX threads

**Comment by ericLemanissier** [comment]
I just realized the tests are not compiled in this configuration. I am currently trying with pseudo targets and pseudo dependencies ON, because it seems mandatory to compile all the tests. Please don't merge the current version of the pull request

**Comment by hkaiser** [comment]
In order to run the tests you'll have to explicitly request building them:

    make -jN tests

**Comment by ericLemanissier** [comment]
Yes, but the "tests" target does not exist if HPX_WITH_PSEUDO_DEPENDENCIES is OFF
The error was the following:

```
CMake Warning (dev) at CMakeLists.txt:1344 (add_custom_command):
  Policy CMP0040 is not set: The target in the TARGET signature of
  add_custom_command() must exist.  Run "cmake --help-policy CMP0040" for
  policy details.  Use the cmake_policy command to set the policy and
  suppress this warning.

  The target name "tests" is unknown in this context.
```

**Comment by hkaiser** [comment]
You're right, we disable pseudo dependencies for MSVC by default. This has to be fixed in the cmake files (see https://github.com/STEllAR-GROUP/hpx/blob/master/CMakeLists.txt#L154).

**Comment by ericLemanissier** [comment]
yes, I already made this change (not pushed yet), but I have now some errors in tests cmakefiles. I have no more time today,i will correct it tomorrow

**Comment by ericLemanissier** [comment]
just for information, I am compiling with the following options:
-DHPX_WITH_TOOLS=ON
-DHPX_WITH_PSEUDO_DEPENDENCIES=ON
-DCMAKE_CXX_FLAGS="-DBOOST_NO_AUTO_PTR -DBOOST_SYSTEM_NO_DEPRECATED -D_WIN32_WINNT=0x0601"

with Boost version: 1.58.0
and mingw-w64\x86_64-5.1.0-posix-seh-rt_v4-rev0

**Comment by hkaiser** [comment]
@ericLemanissier What's the state of this PR? Can we go ahead and consider merging it now?

**Comment by ericLemanissier** [comment]
except for #1802 it seems to be ok in the configuration I tested ( i686-5.1.0-posix-dwarf-rt_v4-rev0 and x86_64-5.1.0-posix-seh-rt_v4-rev with boost 1.58 and hwloc 1.11.0). According to me It can be merged

**Comment by hkaiser** [comment]
Thanks! LGTM.

**Pull request merged**